tests/wpt/web-platform-tests/websockets/opening-handshake/003-sets-origin.worker.js

```diff
@@ -0,0 +1,17 @@
+importScripts("/resources/testharness.js");
+importScripts('../constants.js?pipe=sub');
+importScripts('../websocket.js?pipe=sub');
+
+async_test(function(t) {
+  var ws = new WebSocket(SCHEME_DOMAIN_PORT+'/origin');
+  ws.onmessage = t.step_func(function(e) {
+    assert_equals(e.data, location.protocol+'//'+location.host);
+    ws.onclose = t.step_func(function(e) {
+      assert_equals(e.wasClean, true);
+      ws.onclose = t.step_func(function() {assert_unreached()});
+      setTimeout(t.step_func_done(function() {done();}), 50)
```

Comment:
Just put the `done()` call as the final line of the file; it signals that all the `async_test`s have been defined, not that they have all finished. That's what the `t.done()` call here does.